benchmarks/src/charconv_floats.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+#include <charconv>
+#include <random>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <system_error>
+#include <type_traits>
+#include <vector>
+
+using namespace std;
+
+void verify(const bool b) {
+    if (!b) {
+        puts("FAIL");
+        exit(EXIT_FAILURE);
+    }
+}
+
+enum class RoundTrip { Sci, Fix, Gen, Hex, Lossy };
+
+constexpr chars_format chars_format_from_RoundTrip(const RoundTrip rt) {
+    switch (rt) {
+    case RoundTrip::Sci:
+        return chars_format::scientific;
+    case RoundTrip::Fix:
+        return chars_format::fixed;
+    case RoundTrip::Gen:
+        return chars_format::general;
+    case RoundTrip::Hex:
+        return chars_format::hex;
+    case RoundTrip::Lossy:
+    default:
+        puts("FAIL");
+        exit(EXIT_FAILURE);
+    }
+}
+
+template <RoundTrip Rt, typename Floating, typename... Args>
+void test_to_chars(benchmark::State& state, const Args&... args) {
+    constexpr size_t n = 2'000'000; // how many floating-point values to test
+
+    constexpr size_t BufSize = 2'000; // more than enough
+
+    mt19937_64 mt64;

```

Comment:
It's not terribly important, but `mt64` isn't needed after preparing the `vector`, so we can reduce its scope.